tests/test_custom_checkers/test_pycodestyle_checker.py

```diff
@@ -1,406 +1,92 @@
 import os
 
 import pylint.testutils
+import pytest
 from astroid.astroid_manager import MANAGER
 
 from python_ta.checkers.pycodestyle_checker import PycodestyleChecker
 
-DIR_PATH = os.path.join(__file__, "../../../examples/custom_checkers/e9989_pycodestyle/")
+DIR_PATH = os.path.normpath(
+    os.path.join(__file__, "../../../examples/custom_checkers/e9989_pycodestyle/")
+)
+
+# Define parameter sets for different error and no-error tests
+error_params = [
+    ("e115_error.py", "E115", 2, 0, "expected an indented block (comment)"),
+    ("e122_error.py", "E122", 2, 0, "continuation line missing indentation or outdented"),
+    (
+        "e123_error.py",
+        "E123",
+        3,
+        4,
+        "closing bracket does not match indentation of opening bracket's line",
+    ),
+    ("e125_error.py", "E125", 2, 4, "continuation line with same indent as next logical line"),
+    ("e127_error.py", "E127", 2, 19, "continuation line over-indented for visual indent"),
+    ("e129_error.py", "E129", 4, 4, "visually indented line with same indent as next logical line"),
+    ("e131_error.py", "E131", 4, 8, "continuation line unaligned for hanging indent"),
+    ("e203_error.py", "E203", 1, 30, "whitespace before ':'"),
+    ("e222_error.py", "E222", 1, 3, "multiple spaces after operator"),
+    ("e223_error.py", "E223", 1, 1, "tab before operator"),
+    ("e224_error.py", "E224", 1, 3, "tab after operator"),
+    ("e226_error.py", "E226", 1, 5, "missing whitespace around arithmetic operator"),
+    ("e227_error.py", "E227", 1, 5, "missing whitespace around bitwise or shift operator"),
+    ("e228_error.py", "E228", 1, 5, "missing whitespace around modulo operator"),
+    ("e262_error.py", "E262", 1, 7, "inline comment should start with '# '"),
+    ("e265_error.py", "E265", 1, 0, "block comment should start with '# '"),
+    ("e266_error.py", "E266", 1, 0, "too many leading '#' for block comment"),
+    ("e275_error.py", "E275", 1, 16, "missing whitespace after keyword"),
+    ("e301_error.py", "E301", 5, 4, "expected 1 blank line, found 0"),
+    ("e303_error.py", "E303", 6, 0, "too many blank lines (3)"),
+    ("e304_error.py", "E304", 12, 0, "blank lines found after function decorator"),
+]
+
+no_error_params = [
```

Comment:
Another simplification: given that every error code from above has a corresponding "no error" file, you should be able to use a comprehension to compute this list from the first one, rather than manually writing out all error files.